Transfer feature hangs on asset transfer

<!--
Hello 👋 Thank you for submitting an issue.

Before you start, please make sure your issue is understandable and reproducible.
To make your issue readable make sure you use valid Markdown syntax.

https://guides.github.com/features/mastering-markdown/

Please ensure you have also read and understand the contributing guide.

https://github.com/strapi/strapi/blob/main/CONTRIBUTING.md#reporting-an-issue
-->

## Bug report

### Required System information

<!-- Please ensure you are using the Node LTS version (v14 or v16 or v18) -->
<!-- Strapi v3 is not supported unless it is a critical/high security issue -->

- Node.js version: 16+
- NPM version: 8+
- Strapi version: 4.9.2
- Is your project Javascript or Typescript: JS

### Describe the bug

A clear and concise description of what the bug is.

Edit: Based on issue #16437 this does not seem to be limited to cloud.

When attempting to transfer from a local Strapi instance to a Strapi Cloud project, using:

`yarn strapi transfer --to <url> --to-token <token> `
or
`npm run strapi transfer --to <url> --to-token <token>`

The transfer hangs when it gets to the `assets` transfer. Only killing the process will stop it, which then produces an error.

### Steps to reproduce the behavior

1. Go to a local Strapi project, running on 4.8+
2. Login to Strapi Cloud and ensure both local and cloud projects have same schema
3. Try to transfer from local to cloud using either:
`yarn strapi transfer --to <url> --to-token <token> `
or
`npm run strapi transfer --to<url> --to-token <token>`
4. See error

### Expected behavior

That the transfer would go through successfully. 

### Screenshots

![image](https://user-images.githubusercontent.com/43022248/233807422-1f4baf7d-b5f0-45fd-923b-b8ceff93be50.png)


![image](https://user-images.githubusercontent.com/43022248/233736788-4105482f-aa10-4642-bbf9-2a829a6adc61.png)

I'm also experiencing this issue, with 4.11.4!

Client:
```
✔ entities: 109 transfered (size: 156.3 KB) (elapsed: 2571 ms)
✖ assets: 32 transfered (size: 100.8 MB) (elapsed: 27031 ms)
[2023-07-10 16:30:36.663] error: [ERROR] WebSocket is not open: readyState 3 (CLOSED)
Transfer process failed.
```

Server:
```
2023-07-10T14:30:22.047 app[9080559f545018] cdg [info] Error [ERR_HTTP_REQUEST_TIMEOUT]: Request timeout
2023-07-10T14:30:22.047 app[9080559f545018] cdg [info] at new NodeError (node:internal/errors:399:5)
2023-07-10T14:30:22.047 app[9080559f545018] cdg [info] at onRequestTimeout (node:_http_server:787:30)
2023-07-10T14:30:22.047 app[9080559f545018] cdg [info] at Server.checkConnections (node:_http_server:600:7)
2023-07-10T14:30:22.047 app[9080559f545018] cdg [info] at listOnTimeout (node:internal/timers:569:17)
2023-07-10T14:30:22.047 app[9080559f545018] cdg [info] at process.processTimers (node:internal/timers:512:7)
```

--- a/packages/core/data-transfer/src/strapi/providers/local-source/assets.ts
+++ b/packages/core/data-transfer/src/strapi/providers/local-source/assets.ts
@@ -29,7 +29,7 @@ function getFileStream(
 
       if (res.body) {
         // pipe the image data
-        Readable.fromWeb(new webStream.ReadableStream(res.body)).pipe(readableStream);
+        Readable.fromWeb(res.body as any).pipe(readableStream);
       } else {
         readableStream.emit('error', new Error('Empty data found for file'));
       }

--- a/packages/core/data-transfer/src/strapi/providers/local-source/assets.ts
+++ b/packages/core/data-transfer/src/strapi/providers/local-source/assets.ts
@@ -1,6 +1,5 @@
 import { join } from 'path';
 import { Duplex, PassThrough, Readable } from 'stream';
-import * as webStream from 'stream/web';
 import { stat, createReadStream, ReadStream } from 'fs-extra';
 import type { Core } from '@strapi/types';
 

--- a/packages/core/data-transfer/src/strapi/providers/local-source/assets.ts
+++ b/packages/core/data-transfer/src/strapi/providers/local-source/assets.ts
@@ -1,6 +1,7 @@
 import { join } from 'path';
 import { Duplex, PassThrough, Readable } from 'stream';
 import { stat, createReadStream, ReadStream } from 'fs-extra';
+import * as webStream from 'stream/web';
 import type { Core } from '@strapi/types';
 
 import type { IAsset, IFile } from '../../../../types';
@@ -28,7 +29,7 @@ function getFileStream(
 
       if (res.body) {
         // pipe the image data
-        Readable.fromWeb(res.body as any).pipe(readableStream);
+        Readable.fromWeb(res.body as webStream.ReadableStream<Uint8Array>).pipe(readableStream);
       } else {
         readableStream.emit('error', new Error('Empty data found for file'));
       }

